File: hw/marquez/u3/hw30/Strategy/src/main/java/controller/SortingContext.java
package controller;

import model.BubbleSort;
import model.InsertionSort;
import model.QuickSort;

/**
 *
 * @author Jenniffer Marquez 
 */

public class SortingContext {
    private SortingStrategy ss;

    public int[] sort(int data[]) {
        int size = data.length;
        ss = setSortStrategy(size);
        return ss.sort(data);
    }

    // Method to choose the sorting strategy based on the number of elements
    public SortingStrategy setSortStrategy(int n) {
        if (n > 0 && n < 30) {
            ss = new BubbleSort();  // Use BubbleSort for 1 to 29 elements
        } else if (n >= 30 && n < 100) {
            ss = new InsertionSort();  // Use InsertionSort for 30 to 99 elements
        } else if (n >= 100) {
            ss = new QuickSort();  // Use QuickSort for 100 or more elements
        }
        return ss;
    }
}

File: exams/pantoja/u3/Strategy/src/main/java/ec/espe/edu/strategy/model/SortingContext.py
from bubble_sort import BubbleSort
from insertion_sort import InsertionSort
from quick_sort import QuickSort

class SortingContext:
    def __init__(self):
        self.ss = None

    def sort(self, data):
        size = len(data)
        self.ss = self.set_sort_strategy(size)
        return self.ss.sort(data)

    def set_sort_strategy(self, n):
        if 0 < n < 30:
            self.ss = BubbleSort()
        elif 30 <= n < 100:
            self.ss = InsertionSort()
        elif n >= 100:
            self.ss = QuickSort()
        return self.ss

